contrib/Dockerfile.manylinux

```
@@ -239,13 +275,21 @@ RUN wget --tries=3 --waitretry=5 https://www.mellanox.com/downloads/DOCA/DOCA_v3
 ENV CUDA_PATH=/usr/local/cuda
 
 WORKDIR /workspace
+# =============================================================================
+# Build and install UCX (Unified Communication X) and gdrcopy
+# =============================================================================
+# UCX provides high-performance networking primitives
+# gdrcopy enables GPU memory access from CPU
 
+# Remove any existing UCX installations to avoid conflicts
 RUN rm -rf /usr/lib/ucx
 RUN rm -rf /opt/hpcx/ucx
```

Hadolint (2.14.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)